test/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -172,10 +185,21 @@ const LU_NO_PIVOT = VERSION >= v"1.7.0-DEV.1188" ? NoPivot() : Val(false)
             end
 
             @testset "eigen(::Matrix{$T}) rrule" for T in (Float64, ComplexF64)
-                # get a bit away from zero so don't have finite differencing woes
-                # TODO: this better https://github.com/JuliaDiff/ChainRules.jl/issues/379
-                Random.seed!(1)
-                X = 10 .* (rand(T, n, n) .+ 5.0)
+                # uniform distribution over `(-1, 1)` / `(-1, 1)^2`
+                _rand(T, n...) = rand(T, n...) .* rand(T <: Complex ? [1, im, -1, -im] : [1, -1], n...)
+
+                # make sure that each eigenvector has one clearly defined entry with maximum magnitude
+                # so that the normalization of EVs is well defined
+                V = _rand(T, n, n)
+                for (col, i) in zip(eachcol(V), shuffle(1:n))
+                    col[i] += 3 * (T <: Complex ? cispi(2rand()) : rand([1, -1]))
+                    normalize!(col)
+                end
+
+                # make sure the sorting of eigen values is well defined
+                λ = 10(_rand(T, n) .+ (0:3:3(n-1)))
+
+                X = V * Diagonal(λ) / V
 
```

Comment:
@oxinabox Do you prefer duplicating this here over moving this into its own function?

Reply:
I prefer duplicating.
Though we might like to make a helper for this in ChainRulesTestUtils? at some point
Like how we have `generate_well_conditioned_matrix`

Reply:
I did look into whether this was just a matter of generating a matrix with a low enough condition number, but found that that didn't have much to say about whether that matrix was suitable for the testing `eigen` or not. This likely means that "well conditioned" here very much depends on what you are trying to do with the matrix, so I don't think we can have such a generalized function.

Reply:
actually moving it to its own function here could be good.

Your call

Reply:
Ok, I changed it. Seth agreed to give this another look today or tomorrow, so I'd like to hold off on merging until then.